expandybird/expander/expander.go

```diff
@@ -40,6 +42,67 @@ func NewExpander(binary string) Expander {
 	return &expander{binary}
 }
 
+// NewTemplateFromArchive creates and returns a new template whose content
+// and imported files are read from the supplied archive.
+func NewTemplateFromArchive(name string, r io.Reader, importFileNames []string) (*common.Template, error) {
+	var content []byte
+	imports, err := collectImportFiles(importFileNames)
+	if err != nil {
+		return nil, err
+	}
+
+	tr := tar.NewReader(r)
+	for i := 0; true; i++ {
+		hdr, err := tr.Next()
+		if err == io.EOF {
+			break
+		}
+
+		if err != nil {
+			return nil, err
+		}
+
+		if hdr.Name != name {
+			importFileData, err := ioutil.ReadAll(tr)
+			if err != nil {
+				return nil, fmt.Errorf("cannot read archive file %s: %s", hdr.Name, err)
+			}
+
+			imports = append(imports,
+				&common.ImportFile{
+					Name:    path.Base(hdr.Name),
+					Content: string(importFileData),
+				})
+		} else {
+			content, err = ioutil.ReadAll(tr)
+			if err != nil {
+				return nil, fmt.Errorf("cannot read %s from archive: %s", name, err)
+			}
+		}
+	}
+
+	if len(content) < 1 {
+		return nil, fmt.Errorf("cannot find %s in archive", name)
+	}
+
+	return &common.Template{
+		Name:    name,
+		Content: string(content),
+		Imports: imports,
+	}, nil
+}
+
+// NewTemplateFromReader creates and returns a new template whose content
+// is read from the supplied reader.
+func NewTemplateFromReader(name string, r io.Reader, importFileNames []string) (*common.Template, error) {
+	content, err := ioutil.ReadAll(r)
+	if err != nil {
+		return nil, fmt.Errorf("cannot read archive %s: %s", name, err)
```

Comment:
No, it's sometimes just a flat file. Either way, it's coming from a stream, such as stdin, and passed as a reader, hence the name.